### PR TITLE
Disallow custom pseudo-headers

### PR DESCRIPTION
**Pull request by LuRsT**
I had a look at `hypothesis` and realized that the test that was already present in `TestFilter` is testing the "good case", so I only created a test for an invalid pseudo-header, correct me if I'm wrong, but I found no way to do something like what I did in the test in a `hypothesis` way.

### CODE REVIEW

File: test/test_invalid_headers.py [hunk 1]
```diff
@@ -102,3 +102,8 @@ def test_range_of_acceptable_outputs(self, headers):
             assert headers == h2.utilities.validate_headers(headers)
         except h2.exceptions.ProtocolError:
             assert True
+
+    def test_invalid_pseudo_headers(self):
+        headers = ((b':custom', b'value'))
```
Comment (Lukasa) [inline test/test_invalid_headers.py]: I don't think this actually creates a tuple within a tuple. Consider changing this to a list with one tuple entry. That's why your test is failing.

**Comment by Lukasa** [comment]
One small note regarding the tests.

File: test/test_invalid_headers.py (hunk 1 above)
Comment (LuRsT) [inline test/test_invalid_headers.py]: Damn, fail, I did fix this but then I undid it, I'll fix this :+1:

**Comment by LuRsT** [comment]
I force pushed it, don't know if that's the way to go, but wanted a cleaner commit, could you ask travis-ci to re-check it please?

**Comment by LuRsT** [comment]
Disregard, travis-ci is smart :)

**Comment by Lukasa** [comment]
Hooray! Looks great to me, thank you so much! :sparkles: :cake: :sparkles:

**Comment by LuRsT** [comment]
Great! Thanks for your help @Lukasa, and expect more contributions for me :D

**Pull request merged**